client/app/helpers/DateHelper.js

```diff
@@ -87,6 +88,16 @@ export function sameDay(date, other) {
   return moment(date).day() === moment(other).day();
 }
 
+
+export function roundedTime(date, granularity) {
+  const diff = date.unix() % granularity;
+  if (diff > 60) {
```

Comment:
I don't get the magic number 60 here. There's a test case called "with time almost equal to a multiple of granularity" that covers this. Why are you not using a value like `granularity / 2` instead? If this is intended then I think extracting 60 to a well-named constant would clarify a lot.